kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/logs/TestLogCollector.java

```diff
@@ -0,0 +1,232 @@
+        CURRENT_DATE = dateTimeFormatter.format(LocalDateTime.now());
+    }
+
+    /**
+     * TestLogCollector's constructor
+     */
+    public TestLogCollector() {
+        this.logCollector = defaultLogCollector();
+    }
+
+    /**
+     * Method for creating default configuration of the {@link LogCollector}.
+     * It provides default list of resources into the builder and configures the required {@link KubeClient} and
+     * {@link Kubectl}
+     *
+     * @return  {@link LogCollector} configured with default configuration for the tests
+     */
+    private LogCollector defaultLogCollector() {
+        List<String> resources = new ArrayList<>(List.of(
+                Constants.SECRET.toLowerCase(Locale.ROOT),
+                Constants.DEPLOYMENT.toLowerCase(Locale.ROOT),
+                Constants.CONFIG_MAP.toLowerCase(Locale.ROOT),
+                Constants.SERVICE.toLowerCase(Locale.ROOT),
+                Constants.JOB.toLowerCase(Locale.ROOT),
+                Kafka.RESOURCE_SINGULAR,
+                KafkaNodePool.RESOURCE_SINGULAR));
+
+        return new LogCollectorBuilder()
+                .withKubeClient(new KubeClient())
+                .withKubeCmdClient(new Kubectl())
+                .withRootFolderPath(Environment.CLUSTER_DUMP_DIR)
+                .withNamespacedResources(resources.toArray(new String[0]))
+                .build();
+    }
+
+    /**
+     * Method that checks existence of the folder on specified path.
+     * From there, if there are no sub-dirs created - for each of the test-case run/re-run - the method returns the
+     * full path containing the specified path and index (1).
+     * Otherwise, it lists all the directories, filtering all the folders that are indexes, takes the last one, and returns
+     * the full path containing specified path and index increased by one.
+     *
+     * @param rootPathToLogsForTestCase     complete path for test-class/test-class and test-case logs
+     *
+     * @return  full path to logs directory built from specified root path and index
+     */
+    private Path checkPathAndReturnFullRootPathWithIndexFolder(Path rootPathToLogsForTestCase) {
+        File logsForTestCase = rootPathToLogsForTestCase.toFile();
+        int index = 1;
+
+        if (logsForTestCase.exists()) {
+            String[] filesInLogsDir = logsForTestCase.list();
+
+            if (filesInLogsDir != null && filesInLogsDir.length > 0) {
+                index = Integer.parseInt(
+                        Arrays
+                                .stream(filesInLogsDir)
+                                .filter(file -> {
+                                    try {
+                                        Integer.parseInt(file);
+                                        return true;
+                                    }
+                                    catch (NumberFormatException e) {
+                                        return false;
```

Comment:
If you ever go down this path, your code with fail with an ArrayIndexOutOfBoundsException on line 167.

Reply:
Refactored to avoid that.